webapp/src/components/channel_header_actions/channel_header_menu.tsx

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import JiraIcon from 'components/icon';
+
+export type Props = {
+    open: (channelId: string) => void;
+    channelId: string;
+    isInstanceInstalled: boolean;
+    userConnected: boolean;
+};
+
+export default class ChannelHeaderMenuAction extends React.PureComponent<Props> {
+    handleClick = (): void => {
+        const {isInstanceInstalled, userConnected, open, channelId} = this.props;
+        if (isInstanceInstalled && userConnected) {
+            open(channelId);
+        }
```

Comment:
@mickmister I think I should modify this component a bit more.  Currently, the component shows for all users, with the button action only working if the user is a sysadmin, an instance is installed, and the user is connected.  I feel like the button component shouldn't be visible at all instead unless the user clicking it will be shown the subscriptions modal.

Reply:
I agree, the subscribe button should only be visible for users that are able to use the feature

Reply:
This is something we need to ask the server to find out. We would need to ask about the channel we're currently in every time. I wonder if it is inappropriate for a plugin to hit one of its endpoints every time they navigate to a channel. As long as it's efficient it shouldn't matter, but it would show in the dev tools a lot for example.